src/app/shared/components/template/components/data-items/data-items.component.ts

```diff
@@ -80,6 +80,14 @@ export class TmplDataItemsComponent extends TemplateBaseComponent implements OnD
     // TODO - deep diff and only update changed
     this.itemRows = replacedActionRows;
     this.cdr.markForCheck();
+    // Without this second check, and without the delay, templates nested inside data-items loops do not render.
```

Comment:
@chrismclarke – any ideas what could be causing this?

Demo of nested template only rendering after second check:

https://github.com/IDEMSInternational/parenting-app-ui/assets/64838927/1042772a-46e1-4b33-bba0-31da7fcaf802


Another solution, which causes the nested template to render immediately, is to reset the change detection strategy to the default (currently `OnPush`). Am I right in thinking that we use `ChangeDetectionStrategy.OnPush` to avoid unnecessary checking for changes and DOM updates?

Reply:
Could you make a more minimal debug sheet? I tried loading the task group stepper but needs extra data from `feat_task_group_1_data` so hard to debug efficiently

Reply:
Sure, I've made a debug sheet here: [debug_data_items_rendering](https://docs.google.com/spreadsheets/d/14XXwZ7R_7qIBo5xKYuxnPVQbQidI--PPJ7TVvZxwy1k/edit#gid=1422144003).

Here's what I get without the second, delayed mark for check:

https://github.com/IDEMSInternational/parenting-app-ui/assets/64838927/b0e821df-0437-49a1-ba02-08acaa0d5e83

With the second, delayed mark for check:

https://github.com/IDEMSInternational/parenting-app-ui/assets/64838927/36875f04-8837-45bf-9d09-d97f23f777b6

And without `changeDetection: ChangeDetectionStrategy.OnPush` (and no second mark for check):

https://github.com/IDEMSInternational/parenting-app-ui/assets/64838927/608dcca6-7559-4948-933e-7c0cc2f18054

Reply:
Potentially resolved with https://github.com/IDEMSInternational/parenting-app-ui/pull/2221